File: part-2-simple-ut-particle/python/test_particle-correction.py
from unittest import TestCase
from particle import Particle
import pytest

def test_constructor():
	# setup
	particle = Particle(0, 1)

	# test
	assert 0 == particle.get_x()
	assert 1 == particle.get_vx()

def test_move():
	# setup
	particle = Particle(0, 1)

	# action
	particle.move(0.5)

	# check
	assert 0.5 == particle.get_x()
	assert 1 == particle.get_vx(), "Un message pour expliquer"

File: part-2-simple-ut-particle/cpp/test-particle-correction.cpp
/*********************************************************/
#include "particle.hpp"
#include <gtest/gtest.h>

/*********************************************************/
using namespace experiment;

/*********************************************************/
TEST(TestParticle, constructor)
{
	Particle particle(1.0, 1.5);
	EXPECT_EQ(1.0, particle.get_x());
	EXPECT_EQ(1.5, particle.get_vx());
}

/*********************************************************/
TEST(TestParticle, move)
{
	Particle particle(0, 1);
	particle.move(0.5);
	EXPECT_EQ(0.5, particle.get_x());
	EXPECT_EQ(1, particle.get_vx());
}

